### PR TITLE
Feature/core 1092 persist did doc

### PR DESCRIPTION
**Pull request by datrinh**

### CODE REVIEW

File: dapps/profile.vue/src/components/did/DID.ts [hunk 1]
```diff
@@ -19,49 +19,145 @@
...
-    this.endpoints = await DidService.getServiceEndpoints(this.didDocument);
-    console.log('this.didDocument', this.didDocument);
-    console.log('this.delegates', this.delegates);
-    /* // TODO: switch after complete identity switch to: runtime.did.getDidDocument();
-       const identity = await this.runtime.verifications.getIdentityForAccount(this.runtime.activeAccount, true);
-       const did = await this.runtime.did.convertIdentityToDid(identity);
-       const document = await this.runtime.did.getDidDocumentTemplate();
-       await this.runtime.did.setDidDocument(did, document);
-       const retrieved = await this.runtime.did.getDidDocument(did); */
+    this.didService = new DidService(this.getRuntime());
```
Comment (Tschuck) [inline dapps/profile.vue/src/components/did/DID.ts]: Please check at the beginning, if a new identity based profile is loaded. If not, show a message, that the did functionality is not supported with old profiles.

File: dapps/profile.vue/src/components/did/DidService.ts [hunk 2]
```diff
@@ -1,93 +1,124 @@
...
+      service: endpoints.map((endpoint) => ({
+        id: endpoint.label,
+        serviceEndpoint: endpoint.url,
+        type: 'TODO', // TODO
+      })),
+    };
+  }
+
+  /**
+   * TODO: Currently we don't implement controller logic. Return empty array
```
Comment (Tschuck) [inline dapps/profile.vue/src/components/did/DidService.ts]: Why we don't implement controller logic? ;)

**Comment by Tschuck** [comment]
I can add the same delegate multiple times:

image

File: dapps/profile.vue/src/components/did/DidService.ts (hunk 2 above)
Comment (datrinh) [reply dapps/profile.vue/src/components/did/DidService.ts]: We agreed on it during the planning. Currently we don't plan to have multiple controllers I suppose, but I don't really remember. See the note in figma :)

**Comment by datrinh** [comment]
@Tschuck Regarding the validation of inputs: The DID Spec and our backend don't restrict the content of the Delegators and Endpoints. That's what I have discussed with Phil. We usually reflect all possibilities that the backend has :)

**Comment by datrinh** [comment]
> * Please add a check for valid URL input
> * Am I allowed to save empty URLs?
> 
> image

**Comment by datrinh** [comment]
> After I saved a did and reloaded the page, i get the following error in console and a endless loading spinner is shown. Please add a error handling page and check, whats wrong with the did.
> 
> image

It's a bug in the API, that should be fixed by Phil :)

**Comment by Tschuck** [comment]
Okay, maybe we should implement the url checks for the service endpoints for both, front- and backend. You can hit Phil to do it :D

**Comment by datrinh** [comment]
After fiddling with form input validation for a while, I would like to exclude the validation until we find a proper solution to validate single inputs.
It's also not part of the AC, nor validated on the backend. Let's stick with it for the MVP.

**Pull request merged**